pkg/segment/segment.go

```diff
@@ -258,14 +268,28 @@ func IsTelemetryEnabled(cfg preference.Client) bool {
 	klog.V(4).Info("Checking telemetry enable status")
 	// The env variable gets precedence in this decision.
 	// In case a non-bool value was passed to the env var, we ignore it
+
+	//lint:ignore SA1019 We deprecated this env var, but until it is removed, we still need to support it
 	disableTelemetry, _ := strconv.ParseBool(os.Getenv(DisableTelemetryEnv))
-	if disableTelemetry {
-		klog.V(4).Infof("Sending telemetry disabled by %s=%t\n", DisableTelemetryEnv, disableTelemetry)
+	trackingConsent := os.Getenv(TrackingConsentEnv)
+
+	if disableTelemetry || trackingConsent == "no" {
+		klog.V(4).Info("Sending telemetry disabled by env variable\n")
 		return false
-	} else if cfg.GetConsentTelemetry() {
+	}
+
+	if trackingConsent == "yes" {
```

Comment:
Could we have functions `trackingConsentEnabled`/`trackingConsentDisabled` instead, so we could rename the values or add aliases later?

Reply:
Done - added in [`48669d3` (#6258)](https://github.com/redhat-developer/odo/pull/6258/commits/48669d394a74b0ee02bb060080aed800cb5d5c4a).

This made me realize that I had few cases not tested, e.g., when the variables are not present in the env. So I added more unit tests in [`75f2321` (#6258)](https://github.com/redhat-developer/odo/pull/6258/commits/75f23217378f3f83d1b38b11c49611ebbcec41a4)